LBH.AdultSocialCare.Api/V1/Gateways/DayCarePackageOpportunityGateways/DayCarePackageOpportunityGateway.cs

```diff
@@ -0,0 +1,103 @@
+using LBH.AdultSocialCare.Api.V1.Domain.DayCarePackageOpportunityDomains;
+using LBH.AdultSocialCare.Api.V1.Infrastructure.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using LBH.AdultSocialCare.Api.V1.Exceptions;
+using LBH.AdultSocialCare.Api.V1.Factories;
+using LBH.AdultSocialCare.Api.V1.Infrastructure;
+using Microsoft.EntityFrameworkCore;
+
+namespace LBH.AdultSocialCare.Api.V1.Gateways.DayCarePackageOpportunityGateways
+{
+    public class DayCarePackageOpportunityGateway : IDayCarePackageOpportunityGateway
+    {
+        private readonly DatabaseContext _dbContext;
+        private readonly IMapper _mapper;
+
+        public DayCarePackageOpportunityGateway(DatabaseContext dbContext, IMapper mapper)
+        {
+            _dbContext = dbContext;
+            _mapper = mapper;
+        }
+        public async Task<Guid> CreateDayCarePackageOpportunity(DayCarePackageOpportunity dayCarePackageOpportunity)
+        {
+            var entry = await _dbContext.DayCarePackageOpportunities.AddAsync(dayCarePackageOpportunity).ConfigureAwait(false);
+            try
+            {
+                await _dbContext.SaveChangesAsync().ConfigureAwait(false);
```

Comment:
May be good to check for success of save here, as SaveChangesAsync should return 1 for 1 entity saved